### ISSUE TITLE
Workflow does not complete at times

### ISSUE THREAD

**Issue by issam-chalil** [danielgerlag/workflow-core#794]
**Describe the bug**

We are using WorkflowCore in a distributed environment to run sagas. We are using Postgres for persistence and Redis for distributed locking and queuing. Workflows that are started do not complete at times. We see a lot of log messages stating that the hung Worflows are part of a greylist and do not complete, even though all the steps complete without any exceptions.

**To Reproduce**

It seems the the last step in the Workflow is causing an issue. The last step generates a bunch of Url which are then persisted to the database as a part of the workflow data. This step gets completed, but the Workflow never completes. The "StepCompleted" event is triggered, but not the "WorkflowCompleted" event.

**Expected behavior**

Workflows should complete and the "WorkflowCompleted" event should be triggered.

**Additional context**

If the step involving the Url generation is removed the Workflow completes as expected. I have attached a sample app to demonstrate what we are doing in the last step.
[WorkflowCoreSampleApp.zip](https://github.com/danielgerlag/workflow-core/files/6187984/WorkflowCoreSampleApp.zip)

We have tried running the Workflow core in a single node environment with and without Redis. This does not solve the issue in any way. The only thing that gets the Workflow to complete is to remove the final Url generation step.

**Comment by danielgerlag** [comment]
Is the step generating the urls a heavy/long process?

**Comment by issam-chalil** [comment]
No, the step at the most takes around 30-45 seconds to complete.

**Comment by danielgerlag** [comment]
What is your poll interval set to?

**Comment by issam-chalil** [comment]
Which polling interval are you referring to? We haven't explicitly set any polling interval for Workflow Core.

**Comment by wfcore-manager** [comment]
Closing as not planned — this has been open for several years without recent progress. Please open a new issue if this is still relevant on a current Workflow Core release.
